src/model/DatabaseService.php

```diff
@@ -28,6 +28,6 @@ public function __construct($dataBase)
 			echo '<b>Erreur de connexion à la base de données : <br> Ligne : ' . $e->getLine() . ' :</b> ' . $e->getMessage();
 			exit;
 		}
-		$this->bdd = $bdd;
+		return $this->bdd = $bdd;
```

Comment:
Attention ! Il n' y a pas de return dans un constructeur. 
Tu dois déplacer tout le code  du `try catch` dans une fonction de ta classe. Cette fonction peut-être static. (Tu peux regarder le cours d'opencalssrooms sur la programmation orientée objet : https://openclassrooms.com/fr/courses/1665806-programmez-en-oriente-objet-en-php?status=published) et ici du coup tu n'auras plus besoin de ton constructeur.

- Dans ton constructeur tu set tous les éléments de ton tableau  `$database` (dbServer, dbName, dbUser....)  un à un Exemple : `$this->server = $database['dbServer']` (à faire également pour les autres variables de` $database`). Et je serais toi je splitterais mon tableau $database en plusieurs variables dans les paramètres de mon constructeurs pour être sûr que tous les éléments dont tu as besoins pour faire ta connexion sont bien présent)

```suggestion
public function __construct (string $serverName, string $dbName, string $dbUser ....) {
       $this->server = $serverName;
       $this->database = $dbName;
       ........
}
		
```

- Tu développes une fonction **créer la connexion** et qui **retourne une instance de cette connexion** ( `return $pdo`). En gros ton `try catch`

Reply:
Essais de faire au mieux.